tests/it/variant.rs

```diff
@@ -0,0 +1,145 @@
+#![cfg(feature = "time")]
+
+use serde::{Deserialize, Serialize};
+use time::Month::January;
+
+use clickhouse::Row;
+
+// See also: https://clickhouse.com/docs/en/sql-reference/data-types/variant
+
+#[tokio::test]
+async fn variant_data_type() {
+    let client = prepare_database!();
+
+    // NB: Inner Variant types are _always_ sorted alphabetically,
+    // and should be defined in _exactly_ the same order in the enum.
+    #[derive(Debug, PartialEq, Serialize, Deserialize)]
+    enum MyRowVariant {
+        Array(Vec<i16>),
+        Boolean(bool),
+        // attributes should work in this case, too
+        #[serde(with = "clickhouse::serde::time::date")]
+        Date(time::Date),
+        FixedString([u8; 6]),
+        Float32(f32),
+        Float64(f64),
+        Int128(i128),
+        Int16(i16),
+        Int32(i32),
+        Int64(i64),
+        Int8(i8),
+        String(String),
+        UInt128(u128),
+        UInt16(i16),
+        UInt32(u32),
+        UInt64(u64),
+        UInt8(i8),
+    }
+
+    #[derive(Debug, PartialEq, Row, Serialize, Deserialize)]
+    struct MyRow {
+        var: MyRowVariant,
+    }
+
+    // No matter the order of the definition on the Variant types, it will always be sorted as follows:
+    // Variant(Array(UInt16), Bool, FixedString(6), Float32, Float64, Int128, Int16, Int32, Int64, Int8, String, UInt128, UInt16, UInt32, UInt64, UInt8)
+    client
+        .query(
+            "
+            CREATE OR REPLACE TABLE test_var
+            (
+                `var` Variant(
+                    Array(UInt16),
+                    Bool,
+                    Date,
+                    FixedString(6),
+                    Float32, Float64,
+                    Int128, Int16, Int32, Int64, Int8,
+                    String,
+                    UInt128, UInt16, UInt32, UInt64, UInt8
+                )
+            )
+            ENGINE = MergeTree
+            ORDER BY ()",
+        )
+        .with_option("allow_experimental_variant_type", "1")
+        .with_option("allow_suspicious_variant_types", "1")
+        .execute()
+        .await
+        .unwrap();
+
+    let rows = vec![
```

Comment:
It will be x3 shorter with simple `let row = |var|  MyRow { var }` or `[MyRowVariant::...].into_iter().map(|var| MyRow { var }).collect()`